### ISSUE TITLE
[MA0076] Add exclude list to analyzer

### ISSUE THREAD

**Issue by vchirikov** [meziantou/Meziantou.Analyzer#798]
Hi, first of all, thanks for your awesome analyzers! 

**Feature request**

I have a type which represents uuid v7, the type of course implements [`ISpanFormattable`](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/ISpanFormattable.cs,7) which inherits from [`IFormattable`](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/IFormattable.cs,8) The implementation doesn't use `IFormatProvider` at all, but produce false-positive errors. I don't want to disable `ToString` analyzing in whole solution because it very useful for some types (e.g. float/double with `.`/`,`).

```cs
  public string ToString(string? format, IFormatProvider? formatProvider) => ToString();
```

**Describe the solution you'd like**

It will be nice to have an option to exclude a type from analyzer error via .editorconfig.

**Comment by meziantou** [comment]
Which rules should be configurable?

**Comment by vchirikov** [comment]
I think these should be enough:

* [MA0075](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0075.md)
* [MA0076](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0076.md)
* [MA0011](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0011.md)

Also I note that MS provides a similar analyzer `CA1305` (btw with [`excluded_symbol_names`](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1305#exclude-specific-types-and-their-derived-types)). Maybe some of these analyzers might be a duplicate of their.

**Comment by meziantou** [comment]
The rules are now customizable using .NET attributes. You can use the Meziantou.Analyzer.Annotations package or [copy the attribute](https://github.com/meziantou/Meziantou.Analyzer/blob/d965b7c5540261b9162ec70bfec42dc9a6848c38/src/Meziantou.Analyzer.Annotations/CultureInsensitiveTypeAttribute.cs) directly to your project.

Then, you can annotate a type like:

````c#
// For types you don't own
[assembly: Meziantou.Analyzer.Annotations.CultureInsensitiveTypeAttribute(typeof(Foo))]

// On your types
[Meziantou.Analyzer.Annotations.CultureInsensitiveTypeAttribute]
class Bar { }
````

**Comment by vchirikov** [comment]
@meziantou awesome, thanks
